meson.build

```diff
@@ -10,8 +10,12 @@ static_deps = get_option('android') or get_option('default_library') == 'static'
 
 thread_dep = dependency('threads')
 libicu_dep = dependency('icu-i18n', static:static_deps)
-libzim_dep = dependency('libzim', version : '>=3.0.0', static:static_deps)
-pugixml_dep = dependency('pugixml', static:static_deps)
+libzim_dep = dependency('libzim', version : '>=3.0.0', static:static_deps, fallback:['libzim', 'libzim_dep'])
+pugixml_dep = dependency('pugixml', static:static_deps, required:false)
+if not pugixml_dep.found()
+  # FIXME: add static: kwarg when https://github.com/mesonbuild/meson/issues/328 is fixed
+  pugixml_dep = compiler.find_library('pugixml')
+endif
```

Comment:
We are building static binaries in https://github.com/kiwix/kiwix-tools (also using meson).
We do this to publish "autonomous" binaries users just have to run without bothering with extra files or libraries.

If pugixml static is not found by meson at `kiwix-lib` compilation, it will probably not be found at `kiwix-tools` compilation neither.
So it seems to me as a unnecessary workaround here.

Reply:
This change is not about static libraries though; the problem is that libpugixml-dev on Fedora does not ship pkg-config files, so `dependency()` doesn't work, and we have to encode a fallback with `find_library()`. The FIXME is because a missing feature means we can't yet tell meson to force static compilation with find_library().